### ISSUE TITLE
Environment variable is not read

### ISSUE THREAD

**Issue by paloliska** [quarkusio/quarkus#30370]
### Describe the bug

If config property contains dash '-' envornment variable is not mapped to this config property.

### Expected behavior

environment variable mapped to config property

### Actual behavior

..is not mapped

### How to Reproduce?

Reproducer: [repository](https://github.com/paloliska/env-variable)

Steps:
1. clone repository
2. run command as in README.md
3. check logs for printed config values

### Output of `uname -a` or `ver`

Linux xxx 5.15.0-58-generic #64-Ubuntu SMP Thu Jan 5 11:43:13 UTC 2023 x86_64 x86_64 x86_64 GNU/Linux

### Output of `java -version`

openjdk version "17.0.5" 2022-10-18 OpenJDK Runtime Environment (build 17.0.5+8-Ubuntu-2ubuntu122.04) OpenJDK 64-Bit Server VM (build 17.0.5+8-Ubuntu-2ubuntu122.04, mixed mode, sharing)

### Quarkus version or git rev

2.15.3.Final

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

Apache Maven 3.6.3

**Comment by speter555** [comment]
In my opinion, the reason is in this text: https://github.com/eclipse/microprofile-config/blob/master/spec/src/main/asciidoc/configsources.asciidoc#default-configsources . Microprofile config doesn't like hyphen "-".

**Comment by paloliska** [comment]
Yes but stil there are rules for mapping: [mapping](https://github.com/eclipse/microprofile-config/blob/master/spec/src/main/asciidoc/configsources.asciidoc#environment-variables-mapping-rules)
